src/validation.cpp

```diff
@@ -210,9 +210,17 @@ class CChainState {
     bool RollforwardBlock(const CBlockIndex* pindex, CCoinsViewCache& inputs, const CChainParams& params) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 } g_chainstate;
 
-
-
-CCriticalSection cs_main;
+/**
+ * Mutex to guard access to validation specific variables, such as reading
+ * or changing the chainstate.
+ *
+ * This may also need to be locked when updating the transaction pool, e.g. on
```

Comment:
We call the "transaction pool" the "mempool" in so many places in this file, I find the usage of "transaction pool" to be odd...